packages/headless/src/features/case-assist/case-assist-analytics-actions.ts

```diff
@@ -31,128 +36,249 @@ export const logCaseNextStage = (
   );
 
 export const logCreateCase = (): CaseAssistAction =>
-  makeCaseAssistAnalyticsAction(
-    'analytics/caseAssist/case/create',
-    (client, state) =>
-      client.logCaseCreated({
+  makeCaseAssistAnalyticsAction({
+    prefix: 'analytics/caseAssist/case/create',
+    __legacy__getBuilder: (client, state) => {
+      return client.logCaseCreated({
         ticket: caseAssistCaseSelector(state),
-      })
-  );
+      });
+    },
+    analyticsType: 'CaseAssist.CreateTicket',
+    analyticsPayloadBuilder: (state): CaseAssist.CreateTicket => {
+      const {id, category, subject, description, productId} =
+        caseAssistCaseSelector(state);
+      return {
+        ticket: {
+          id: id!,
```

Comment:
Why can't caseAssistSelector return with certitude an id?

Reply:
I think we're not sure to always get it back from Salesforce? Might be just because we are too careful xD

Reply:
exactly it's because we are too careful  but realistically there should not really be a situation where the id is not found.